src/js/util/providers/tinami.js

```diff
@@ -0,0 +1,38 @@
+import convert from 'xml-js';
+
+export default function ($) {
+  return {
+    name: 'TINAMI',
+    setting: 'tinami',
+    re: /(?:www.tinami.com\/view|tinami.jp\/)/,
+    default: true,
+    callback: url => {
+      let imageId = url.slice(url.lastIndexOf('/') + 1);
```

Comment:
I'd prefer if you used the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) API to get this. It would be less error-prone.